src/language-server/engine/GraphQLDataSource.ts

```diff
@@ -1,15 +1,37 @@
 import { DataSourceConfig } from "apollo-datasource";
-import { ApolloLink, execute, GraphQLRequest, makePromise } from "apollo-link";
-import { setContext } from "apollo-link-context";
-import { onError } from "apollo-link-error";
-import { createHttpLink } from "apollo-link-http";
-import {
-  ApolloError,
-  AuthenticationError,
-  ForbiddenError,
-} from "apollo-server-errors";
+import { ApolloLink, execute, GraphQLRequest } from "@apollo/client/link/core";
+import { toPromise as makePromise } from "@apollo/client/link/utils";
+import { onError } from "@apollo/client/link/error";
+import { setContext } from "@apollo/client/link/context";
+import { createHttpLink } from "@apollo/client/link/http";
 import to from "await-to-js";
 import { GraphQLError } from "graphql";
+class ApolloError extends Error {
```

Comment:
This whole file is already an inlined & modified version of https://www.npmjs.com/package/apollo-datasource-graphql (not an actual apollo package), and the `ApolloError` from `apollo-server-errors` was not compatible with `graphql` 16, so I replaced these three classes with inline kinda-similar versions.

In a future PR, I'll probably replace this whole file with `@apollo/client`, since all it is used for is make three different queries.